src/LLVM_Output.cpp

```diff
@@ -21,29 +21,36 @@ std::unique_ptr<llvm::raw_fd_ostream> make_raw_fd_ostream(const std::string &fil
     return raw_out;
 }
 
-void emit_file(llvm::Module &module, Internal::LLVMOStream& out, llvm::TargetMachine::CodeGenFileType file_type) {
+void emit_file(const llvm::Module &module_in, Internal::LLVMOStream& out, llvm::TargetMachine::CodeGenFileType file_type) {
     Internal::debug(1) << "emit_file.Compiling to native code...\n";
-    Internal::debug(2) << "Target triple: " << module.getTargetTriple() << "\n";
+    Internal::debug(2) << "Target triple: " << module_in.getTargetTriple() << "\n";
+
+    // Work on a copy of the module to avoid modifying the original.
+    #if LLVM_VERSION < 38
+    llvm::Module *module = llvm::CloneModule(&module_in);
```

Comment:
Isn't this leaking now?

I'm pretty sure that if you just write this:

    std::unique_ptr<llvm::Module> module(llvm::CloneModule(&module_in));

it would just work on all versions of LLVM, and not leak on old versions of LLVM either.